zeebe/msgpack-value/src/main/java/io/camunda/zeebe/msgpack/property/BaseProperty.java

```diff
@@ -98,16 +103,48 @@ public void write(final MsgPackWriter writer) {
     valueToWrite.write(writer);
   }
 
-  public void writeJSON(final StringBuilder sb) {
+  public void writeJSON(final StringBuilder sb, final boolean maskSanitized) {
     key.writeJSON(sb);
     sb.append(":");
     if (hasValue()) {
-      resolveValue().writeJSON(sb);
+      if (maskSanitized && isSanitized) {
+        sb.append("\"***\"");
+      } else {
+        resolveValue().writeJSON(sb);
+      }
     } else {
       sb.append("\"NO VALID WRITEABLE VALUE\"");
     }
   }
 
+  public boolean isSanitized() {
+    return isSanitized;
+  }
+
+  /**
+   * If set to true, will be replaced by '***' when writing to console or logs, i.e. when called via
+   * {@link #toString()}, or {@link #writeJSON(StringBuilder, boolean)} with the second param set to
```

Comment:
🔧 The comment is a bit misleading. Sounds like for a property which I can set to true or false. But it is the method which I call, to make sure the property IS sanitized.

Proposal:

```
Marks the property as sanitized. During logging, writing to STDOUT, value will be replaced by '***'
```